src/main/java/appeng/crafting/pattern/AECraftingPattern.java

```diff
@@ -363,8 +363,17 @@ public ItemStack getOutput(Container container, Level level) {
         return output;
     }
 
-    public NonNullList<ItemStack> getRemainingItems(CraftingContainer container) {
-        return this.recipe.getRemainingItems(container);
+    public NonNullList<ItemStack> getRemainingItems(Container container) {
+        var nonNullList = NonNullList.withSize(container.getContainerSize(), ItemStack.EMPTY);
+
+        for(int i = 0; i < nonNullList.size(); ++i) {
+            Item item = container.getItem(i).getItem();
+            if (item.hasCraftingRemainingItem()) {
+                nonNullList.set(i, new ItemStack(item.getCraftingRemainingItem()));
+            }
+        }
+
+        return nonNullList;
```

Comment:
No, this is wrong. The entire point of this method is to correctly handle recipes with custom remainder logic. See also 78f5e6d.